### PR TITLE
feat: basic UI for merging sandboxes

### LINKED ISSUE #3436: sandboxes: UI to merge sandboxes
Simple UI:

Image

This is a bit scarier than I want, but as soon as we've put it through QA we'll remove a lot of the scary language.

Right now, in this first pass, I want to be super clear about everything

In this first pass, we are NOT allowing users to select which workflow to merge. Ideally we would also delete the sandbox - but we can push that functionality back if it's more than a function call

### PR DESCRIPTION
**Pull request by elias-ba**
## Description

This PR adds UI for the sandbox merging features to allow users to consolidate their work by merging one sandbox into another. This enables a Git-like workflow where developers can experiment in isolated sandboxes and then merge their changes back into a parent or sibling project.

This PR also fixes a bug discovered during implementation where new workflows lose all their content during merge. When merging a sandbox that contains new workflows (workflows that don't exist in the target project), those workflows were created as empty shells with only their name preserved, all jobs, triggers, and edges were silently lost.

### Key Features
- Merge any sandbox into its parent or sibling projects (not descendants)
- Automatically closes the source sandbox after successful merge
- Merges all workflows from source to target using existing `MergeProjects` module
- Smart target selection with parent project as default
- New merge button with branch icon on each sandbox card
- Modal dialog with clear merge flow and target selection
- Real-time feedback on merge impact and affected sandboxes
- Descendant warnings: Shows which child sandboxes will be closed when merging
- Beta warning: Reminds users this is a beta feature and suggests using CLI for production
- Follows existing sandbox permission model (owners and admins only)

Closes #3436 #3742 #3744

### Basic Merge Flow
1. Create a root project "Production" with one existing workflow
2. Create a sandbox "Feature Test" from Production
3. In the sandbox, create a NEW workflow:
   - Name: "New Feature Workflow"
   - Add a webhook trigger
   - Add 2-3 jobs with code (e.g., `fn(state => state)`)
   - Connect them with edges
   - Verify the workflow canvas shows all components before merging
4. Navigate to Sandboxes page as an owner
5. Click the merge button (branch icon) on "Feature Test" sandbox
6. Verify modal opens with:
   - Source sandbox name displayed
   - "Production" selected by default as target
   - All valid targets listed (parent + siblings, excluding descendants)
7. Submit the merge
8. Verify redirect to Production workflows page
9. Verify success message and "Feature Test" sandbox is deleted
10. Open "New Feature Workflow" in Production:
    - All jobs should be present with their code
    - Webhook trigger should be present
    - All edges should be intact
    
### Descendant Warnings
1. Create a hierarchy: `root -> child1 -> grandchild1`
2. Open merge modal for `child1`
3. Verify single descendant warning appears mentioning `grandchild1`
4. Create another grandchild: `root -> child1 -> grandchild2`
5. Open merge modal for `child1` again
6. Verify multi-descendant warning with full list of both grandchildren

### Deep Nesting
1. Create: `root -> child -> grandchild -> great-grandchild`
2. Open merge modal for `child`
3. Verify all descendants (grandchild + great-grandchild) are listed in warning
4. Verify target dropdown excludes all descendants

### Authorization
1. As a viewer, navigate to Sandboxes page
2. Verify merge button is:
   - Disabled (cursor-not-allowed style)
   - Shows tooltip: "You are not authorized to merge this sandbox"
3. Verify cannot open modal (server rejects with error flash)
4. As an owner, verify merge button is enabled and functional

### Edge Cases
1. No siblings: Open merge modal on an only-child sandbox
   - Verify only parent appears in target dropdown
2. Multiple target options: Verify dropdown shows parent + all siblings
3. Merge error handling: Mock an import failure
   - Verify error message appears
   - Verify modal closes
   - Verify sandbox is not deleted

### Design Decisions

1. Why close descendants? When merging a sandbox, its descendants lose their parent context. Rather than orphaning them or moving them, we close them and warn the user to merge children first if they want to preserve that work.

2. Why only parent/siblings? Merging into a descendant would create a circular dependency in the project hierarchy. This restriction keeps the tree structure valid.

3. Authorization follows update/delete Merge is a destructive operation (deletes source sandbox) and modifies the target, so it requires the same permissions as update/delete operations.

### Known Limitations

- No preview: Users cannot preview changes before merging (CLI workflow recommended for production)
- No undo: Merge is permanent and cannot be rolled back

### Bug Fix Details

The bug was in `Lightning.Projects.MergeProjects.build_merged_workflows/3`. For new workflows, it only copied metadata but ignored all content. This has been fixed by adding a proper `build_new_workflow/1` helper that handles jobs, triggers, edges, and ID remapping correctly.

## AI Usage

Please disclose how you've used AI in this work (it's cool, we just want to know!):

- [x] Code generation (copilot but not intellisense)
- [ ] Learning or fact checking
- [ ] Strategy / design
- [ ] Optimisation / refactoring
- [x] Translation / spellchecking / doc gen
- [ ] Other
- [ ] I have not used AI

You can read more details in our [Responsible AI Policy](https://www.openfn.org/ai#pull-request-templates)

## Pre-submission Checklist

- [x] I have performed a **self-review** of my code.
- [x] I have implemented and tested all related **authorization policies**. (`:owner` and `:admin` can merge; `:viewer` cannot)
- [x] I have updated the **changelog**.
- [x] I have ticked a box in "AI usage" in this PR

### CODE REVIEW

File: test/lightning/projects/merge_projects_test.exs [hunk 1]
```diff
@@ -169,9 +210,12 @@ defmodule Lightning.Projects.MergeProjectsTest do
 
       result = MergeProjects.merge_project(source_project, target_project)
 
-      # Should preserve target ID but use source name
```
Comment (josephjclark) [inline test/lightning/projects/merge_projects_test.exs:160]: Yes, at the PROJECT level this behaviour was wrong.

A the WORKFLOW level it would be correct

Great catch @elias-ba

Comment (josephjclark) [reply test/lightning/projects/merge_projects_test.exs:160]: I like the rewording of the comment too

**Comment by mercy-ship-it** [comment]
QA Verification Checklist: Sandbox Merge Feature
🔹 General Verification
 

- [x] Confirm the new “Merge” (branch icon) button appears on all sandbox cards.
- [x]  Verify the merge modal opens correctly with source and target projects displayed.
- [x]  Confirm only parent and sibling sandboxes appear as merge targets (no descendants).
- [x]  Verify the beta warning message appears in the modal.
- [x]  Confirm source sandbox is automatically deleted after successful merge.
- [x]  Validate redirect to target (Production) project’s workflow page after merge.

🔹 Workflow Merge Validation
 

- [x] Verify new workflows from the source appear in the target with full content (jobs, triggers, edges).
- [x]  Confirm existing workflows are updated correctly without data loss.
- [x]  Check that all UUID mappings are preserved correctly (no duplication).
- [x]  Validate that trigger configurations and job scripts are retained.
- [x]  Confirm workflow connections (edges) remain intact post-merge.

🔹 Descendant & Hierarchy Warnings

- [ ]  Validate descendant warnings appear correctly for nested sandboxes.
- [ ]  Check that all descendants listed in the warning match actual hierarchy.
- [ ]  Confirm target dropdown excludes all descendants.

🔹 Authorization & Permissions
 

- [ ] As viewer, verify the merge button is disabled with correct tooltip.
- [ ]  As admin/owner, verify merge button is enabled and functional.
- [ ]  Check unauthorized users cannot trigger a merge (server should return error flash).

🔹 Error Handling & Edge Cases
 

- [ ] Simulate a merge failure → confirm error message appears and sandbox is not deleted.
- [ ]  Test case with no siblings → only parent shown in dropdown.
- [ ]  Test case with multiple siblings → all valid targets shown.
- [ ]  Check UI and functionality on slow or unstable network.
- [ ]  Confirm no duplicate workflows are created post-merge.

🔹 Regression & Stability Checks
 

- [ ] Verify existing sandbox creation and deletion still function correctly.
- [ ]  Check that merging does not affect unrelated projects.
- [ ]  Validate refresh and redirection behaviors across pages.
- [ ]  Confirm that closed sandboxes are marked appropriately in the UI.

🔹 Bug Fix Validation
 

- [ ] Confirm newly created workflows retain all content after merge (no empty shells).
- [ ]  Validate that jobs, triggers, and edges appear exactly as they were before merge.
